README.md

```diff
@@ -41,6 +43,61 @@ fn main() {
 }
 ```
 
+### Generate a new GPX file
+This example only generates tracks. You can add waypoints and routes as well by instantiating new ``Waypoint``s and ``Route``s.
+```rust
+use std::path::Path;
+use gpx::{Gpx, Track, TrackSegment, Waypoint, Route};
+use geo_types::{Point, coord};
+
+pub fn to_gpx<P: AsRef<Path>>(out_path: P) -> Result<String, Box<dyn Error>> {
+    // Instantiate Gpx struct
+    let mut gpx: Gpx = Gpx {
+        version: GpxVersion::Gpx11, // or Gpx10
+        creator: None, // Option<String>
+        metadata: None, // Option<Metadata>
+        waypoints: vec![], // Vec<Waypoint>
+        tracks: vec![], // Vec<Track>
+        routes: vec![], // Vec<Route>
+    };
+    gpx.tracks.push(Track {
+        name: Some("Track 1".to_string()), // Option<String>
+        comment: None, // Option<String>
+        description: None, // Option<String>
+        source: None, // Option<String>
+        links: vec![], // Vec<Link>
+        type_: None, // Option<String>
+        number: None, // Option<u32>
+        segments: vec![], // Vec<TrackSegment>
+    });
+    gpx.tracks[0].segments.push(TrackSegment { 
+        points: vec![] // Vec<Waypoint>
+    });
+
+    // Create file at path
+    let gpx_file = File::create(out_path);
+    
+    // Add track point
+    let geo_coord = coord! { x: -121.1, y: 38.82 };
+    let geo_point: Point = geo_coord.into();
+    gpx.tracks[0].segments[0].points.push(Waypoint::new(geo_point));
+
+    // Write to file
+    write(&gpx, gpx_file?)?;
+
+    Ok(());
+}
+```
+
+### Write to string
+`write` will write the GPX output to anything that implements `std::io::Write`. To save the output to a string, use a mutable `BufWriter` to write it to a vector, and then convert the vector to a string.
```

Comment:
I don't think you need to wrap a `Vec` in a `BufWriter`. But it might be a good idea to do it for files, since `write` might cause very short writes. So can you update both examples?